examples/workflows/upstream/http_success_condition.py

```diff
@@ -0,0 +1,76 @@
+from hera.workflows import HTTP, Step, Steps, Workflow
+from hera.workflows.models import Arguments, Inputs, Parameter
+
+with Workflow(
+    api_version="argoproj.io/v1alpha1",
+    kind="Workflow",
+    annotations={
+        "workflows.argoproj.io/description": "Exemplifies usage of successCondition in HTTP template (available since v3.3)\n"
+    },
+    generate_name="http-template-condition-",
+    entrypoint="main",
+) as w:
+    with Steps(
+        name="main",
+    ) as invocator:
+        with invocator.parallel():
+            Step(
+                arguments=Arguments(
+                    parameters=[
+                        Parameter(
+                            name="url",
+                            value="http://httpstat.us/201",
+                        )
+                    ],
+                ),
+                name="http-status-is-201-succeeds",
+                template="http-status-is-201",
```

Comment:
Might want to use the same algorithm for sorting parameters that we chose for sorting keys in YAML -- or perhaps copy the field ordering from the types?